x-pack/libbeat/management/input_reload_test.go

```diff
@@ -0,0 +1,219 @@
+			reloadCallCount++
+			if reloadCallCount == 1 {
+				e1 := multierror.Errors{withCause{&common.ErrInputNotFinished{
+					State: "<state string goes here>",
+					File:  "/tmp/foo.log",
+				}}}
+				return e1.Err()
+			}
+
+			return nil
+		},
+	}
+	r.MustRegisterInput(inputs)
+
+	configIdx := -1
+	onObserved := func(observed *proto.CheckinObserved, currentIdx int) {
+		configIdx = currentIdx
+	}
+
+	srv := mockSrv([][]*proto.UnitExpected{
+		{
+			{
+				Id:             "output-unit",
+				Type:           proto.UnitType_OUTPUT,
+				ConfigStateIdx: 1,
+				State:          proto.State_HEALTHY,
+				LogLevel:       proto.UnitLogLevel_DEBUG,
+				Config: &proto.UnitExpectedConfig{
+					Id:   "default",
+					Type: "elasticsearch",
+					Name: "elasticsearch",
+				},
+			},
+			{
+				Id:             "input-unit-1",
+				Type:           proto.UnitType_INPUT,
+				ConfigStateIdx: 1,
+				State:          proto.State_HEALTHY,
+				LogLevel:       proto.UnitLogLevel_DEBUG,
+				Config: &proto.UnitExpectedConfig{
+					Id:   "log-input",
+					Type: "log",
+					Name: "log",
+					Streams: []*proto.Stream{
+						{
+							Id: "log-input-1",
+							Source: requireNewStruct(t, map[string]interface{}{
+								"paths": []interface{}{"/tmp/foo.log"},
+							}),
+						},
+					},
+				},
+			},
+		},
+		{
+			{
+				Id:             "output-unit",
+				Type:           proto.UnitType_OUTPUT,
+				ConfigStateIdx: 1,
+				State:          proto.State_HEALTHY,
+				LogLevel:       proto.UnitLogLevel_DEBUG,
+				Config: &proto.UnitExpectedConfig{
+					Id:   "default",
+					Type: "elasticsearch",
+					Name: "elasticsearch",
+				},
+			},
+			{
+				Id:             "input-unit-1",
+				Type:           proto.UnitType_INPUT,
+				ConfigStateIdx: 1,
+				State:          proto.State_HEALTHY,
+				LogLevel:       proto.UnitLogLevel_DEBUG,
+				Config: &proto.UnitExpectedConfig{
+					Id:   "log-input-2",
+					Type: "log",
+					Name: "log",
+					Streams: []*proto.Stream{
+						{
+							Id: "log-input-2",
+							Source: requireNewStruct(t, map[string]interface{}{
+								"paths": []interface{}{"/tmp/foo.log"},
+							}),
+						},
+					},
+				},
+			},
+		},
+	},
+		[]uint64{1, 1},
+		[]*proto.Features{
+			nil,
+			nil,
+		},
+		onObserved,
+		500*time.Millisecond,
+	)
+	require.NoError(t, srv.Start())
+	defer srv.Stop()
+
+	client := client.NewV2(fmt.Sprintf(":%d", srv.Port), "", client.VersionInfo{
+		Name:    "program",
+		Version: "v1.0.0",
+		Meta: map[string]string{
+			"key": "value",
+		},
+	}, grpc.WithTransportCredentials(insecure.NewCredentials()))
+
+	m, err := NewV2AgentManagerWithClient(
+		&Config{
+			Enabled: true,
+		},
+		r,
+		client,
+		WithChangeDebounce(300*time.Millisecond),
+		WithForceReloadDebounce(800*time.Millisecond),
+	)
+	require.NoError(t, err)
+
+	mm := m.(*BeatV2Manager)
+
+	err = m.Start()
+	require.NoError(t, err)
+	defer m.Stop()
+
+	forceReloadState := []bool{false, true, false}
```

Comment:
[Suggestion]
As it's a slice, use the plural to reflect that
```suggestion
	forceReloadStates := []bool{false, true, false}
```